express-post-json/index.js

```diff
@@ -0,0 +1,29 @@
+const express = require('express');
+const app = express();
+
+var nextId = 1;
+var grades = {};
+
+app.use(express.json());
+
+app.get('/api/grades', (req, res) => {
+  const gradesArr = [];
+  for (const prop in grades) {
+    gradesArr.push(grades[prop]);
+  }
+  res.json(gradesArr);
+});
+
+app.post('/api/grades/', (req, res) => {
+  const postedGrade = req.body;
+  const id = nextId;
+  postedGrade.id = id;
+  grades[id] = postedGrade;
+  res.status(201);
+  res.json(postedGrade);
+  nextId++;
+});
```

Comment:
The response shouldn't be sent until after everything else is updated, otherwise the client may receive a success response when part of the update failed.